core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -1585,53 +1588,7 @@ private UnionOnCalcsToUnionUnifyRule() {
     }
```

Comment:
Why don't we just change this rule to accept different rel types so that we can use 1 class and create 2 instances?

Reply:
Thanks for review, Sorry for the late reply.
`UnionOnCalcsToUnionUnifyRule` and 'IntersectOnCalcsToIntersectUnifyRule' have a clearer meaning. In the process of matching，If we define a rule called `SetOpOnCalcsToSetOpUnifyRule`, We need to judge the matching logic, this should be done by the framework itself.